File: Java/labs/lab5/task_3_2/src/Salesperson.java
public class Salesperson {

  private final String name;

  private final Manager manager;

  public Salesperson(String name, Manager manager) {
    this.name = name;
    this.manager = manager;
  }

  public void payExpenses(int amount) {
    System.out.println(name + " has been paid $" + amount);
  }

}

File: Cpp/lab09_Composite/task_3_2/src/salesperson.cpp
#include <iostream>

#include "salesperson.hpp"


Salesperson::Salesperson(std::string const& name, Manager* manager)
: name(name), manager(manager) {}

void Salesperson::payExpenses(int amount)
{
  std::cout << this->name << " has been paid $" << amount << std::endl;
}
